src/Microsoft.AspNet.Server.Kestrel/Http/MemoryPoolBlock2.cs

```diff
@@ -0,0 +1,551 @@
+                        return -1;
+                    }
+                    else
+                    {
+                        block = block.Next;
+                        index = block.Start;
+                    }
+                }
+            }
+
+            public int Seek(int char0)
+            {
+                if (IsDefault)
+                {
+                    return -1;
+                }
+
+                var byte0 = (byte)char0;
+                var vectorStride = Vector<byte>.Count;
+                var ch0Vector = new Vector<byte>(byte0);
+
+                var block = _block;
+                var index = _index;
+                var array = block.Array;
+                while (true)
+                {
+                    while (block.End == index)
+                    {
+                        if (block.Next == null)
+                        {
+                            _block = block;
+                            _index = index;
+                            return -1;
+                        }
+                        block = block.Next;
+                        index = block.Start;
+                        array = block.Array;
+                    }
+                    while (block.End != index)
+                    {
+                        var following = block.End - index;
+                        if (following >= vectorStride)
+                        {
+                            var data = new Vector<byte>(array, index);
+                            var ch0Equals = Vector.Equals(data, ch0Vector);
+                            var ch0Count = Vector.Dot(ch0Equals, _dotCount);
+
+                            if (ch0Count == 0)
+                            {
+                                index += vectorStride;
+                                continue;
+                            }
+                            else if (ch0Count == 1)
+                            {
+                                _block = block;
+                                _index = index + Vector.Dot(ch0Equals, _dotIndex);
+                                return char0;
+                            }
+                            else
+                            {
+                                following = vectorStride;
+                            }
+                        }
+                        for (; following != 0; following--, index++)
+                        {
+                            if (block.Array[index] == byte0)
+                            {
+                                _block = block;
+                                _index = index;
+                                return char0;
+                            }
+                        }
+                    }
+                }
+            }
+
+            public int Seek(int char0, int char1)
+            {
+                if (IsDefault)
+                {
+                    return -1;
+                }
+
+                var byte0 = (byte)char0;
```

Comment:
yeah :) they actually were byte arguments at one point, but then the caller needed to have `Seek((byte)'\r)` which was nasty... Turns out that '\r' will implicitly become an `int`, so it can `Seek('\r')` now.

The return value being and int is also valuable to be able to return -1, which is outside the set of valid char/byte values. A minor concern, but seemed nicer all things considered...